spyder/preferences/tests/test_shorcuts.py

```diff
@@ -9,29 +9,207 @@
 """
 
 import os
+import sys
 
 # Test library imports
 import pytest
+from qtpy.QtCore import Qt
 
 # Local imports
-from spyder.preferences.shortcuts import ShortcutsTable
+from spyder.config.main import CONF
+from spyder.preferences.shortcuts import (
+    ShortcutsTable, ShortcutEditor, NO_WARNING, SEQUENCE_CONFLICT,
+    INVALID_KEY, SEQUENCE_EMPTY)
 
+
+# ---- Qt Test Fixtures
 @pytest.fixture
-def setup_shorcuts(qtbot):
+def shorcut_table_bot(qtbot):
     """Set up shortcuts."""
-    widget = ShortcutsTable()
-    qtbot.addWidget(widget)
-    return widget
+    shorcut_table = ShortcutsTable()
+    qtbot.addWidget(shorcut_table)
+    return shorcut_table, qtbot
+
+
+@pytest.fixture
+def shortcut_editor_bot(shorcut_table_bot):
+    shorcut_table, qtbot = shorcut_table_bot
+    shortcuts = shorcut_table.source_model.shortcuts
+
+    def _create_bot(context, name):
+        sequence = CONF.get('shortcuts', "{}/{}".format(context, name))
+        shortcut_editor = ShortcutEditor(
+            shorcut_table, context, name, sequence, shortcuts)
+        qtbot.addWidget(shortcut_editor)
+        shortcut_editor.show()
+        return shortcut_editor, qtbot
+    return _create_bot
 
 
-@pytest.mark.skipif(not os.name == 'nt', reason="It segfaults too much on Linux")
-def test_shortcuts(qtbot):
+# ---- Tests ShortcutsTable
+@pytest.mark.skipif(
+    sys.platform.startswith('linux') and os.environ.get('CI') is not None,
+    reason="It fails on Linux due to the lack of a proper X server.")
+def test_shortcuts(shorcut_table_bot):
```

Comment:
This needs to be

```python
def test_shortcuts(shorcut_table_bot, qtbot)
```

and the same for all other tests present in this file.

Reply:
It makes tests easier to read because you can see all fixtures it depends on.

Reply:
ok, I understand. I will change that.

Reply:
Great, thanks!